lib/src/run.dart

```diff
@@ -95,12 +111,14 @@ Future<String> runAsync(String executable,
     broadcastStderr.listen((List<int> data) => stderr.addAll(data));
     stderrLines.listen(logStderr);
 
+    var encoding = runOptions.stdoutEncoding != null
+        ? runOptions.stdoutEncoding : SYSTEM_ENCODING;
     return process.exitCode.then((int code) {
-      var stdoutString = SYSTEM_ENCODING.decode(stdout);
+      var stdoutString = encoding.decode(stdout);
```

Comment:
I think we also need to pass `stdoutEncoding` and `stderrEncoding` to the `toLineStream` method, and use it there.